### PR TITLE
Add tag support to journald logging driver, closes #19556

### LINKED ISSUE #19556: Support tags for journald logging driver
It doesn't seem to be hard to add. If there are no objections on doing this, I can make a PR.

cc @dqminh

### PR DESCRIPTION
**Pull request by bobrik**
Tested manually on debian jessie, it worked :whale:

### CODE REVIEW

**Comment by bobrik** [comment]
Windows build failed, but it does seem like a transient network failure.

File: daemon/logger/journald/journald.go [hunk 1]
```diff
@@ -48,10 +49,17 @@ func New(ctx logger.Context) (logger.Logger, error) {
...
+	tag, err := loggerutils.ParseLogTag(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
 	vars := map[string]string{
 		"CONTAINER_ID":      ctx.ContainerID[:12],
 		"CONTAINER_ID_FULL": ctx.ContainerID,
 		"CONTAINER_NAME":    name,
+		"CONTAINER_TAG":     tag,
```
Comment (dqminh) [inline daemon/logger/journald/journald.go]: SYSLOG_IDENTIFIER is probably more correct as tag

**Comment by calavera** [comment]
Change LGTM

**Comment by runcom** [comment]
LGTM

**Comment by vdemeester** [comment]
@bobrik you need to update the options paragraph of ` docs/reference/logging/journald.md ` too, to add `tag` option (see https://github.com/docker/docker/blob/master/docs/reference/logging/fluentd.md#tag for example)

File: daemon/logger/journald/journald.go (hunk 1 above)
Comment (bobrik) [inline daemon/logger/journald/journald.go]: I want to use `CONTAINER_TAG` to centralize log collection for services. If docker container does not provide tag, I probably don't want to centralize logs from this service. Using `CONTAINER_TAG` allows me to filter by it, while `SYSLOG_IDENTIFIER` probably does not.

It is also more consistent with existing parameters.

**Comment by bobrik** [comment]
@vdemeester updated docs.

**Comment by bobrik** [comment]
Any chance for this to make it in 1.10?

**Comment by vdemeester** [comment]
@bobrik sadly no :sweat_smile: the code freeze was on 14/01/2016 — we're way past that date :sob:

File: daemon/logger/journald/journald.go [hunk 2]
```diff
@@ -48,10 +49,17 @@ func New(ctx logger.Context) (logger.Logger, error) {
 		name = name[1:]
 	}
 
+	// parse log tag
+	tag, err := loggerutils.ParseLogTag(ctx, "")
```
Comment (MHBauer) [inline daemon/logger/journald/journald.go]: Is there any sane default to provide here?

**Comment by MHBauer** [comment]
docs LGTM
- [x] 80 char word wrap
- [x] spelling 
- [x] Markdown
- [x] Sentences end in a period
- [x] code change has something to do docs change

File: daemon/logger/journald/journald.go (hunk 2 above)
Comment (bobrik) [inline daemon/logger/journald/journald.go]: Nope. I think it has to be user provided value so you can ignore logs from containers that don't have it set.

File: docs/reference/logging/journald.md [hunk 3]
```diff
@@ -41,6 +42,11 @@ You can set the logging driver for a specific container by using the
 Users can use the `--log-opt NAME=VALUE` flag to specify additional
 journald logging driver options.
 
+### tag
+
+Template to set `CONTAINER_TAG` value in journald logs, refer to
```
Comment (vdemeester) [inline docs/reference/logging/journald.md]: Hum this one should be a sentence, something like `Specify tag for message, which interpret some markup. Refer to the log tag option documentation for customizing the log tag format.` (borrowed from `splunk.md`).

File: daemon/logger/journald/journald.go (hunk 2 above)
Comment (vdemeester) [inline daemon/logger/journald/journald.go]: I was wondering on this one too (as @MHBauer). For the other logging driver that supports the `tag`, the default is `docker.{{.ID}}` (the container short id). 

Although it duplicate the containerID by default in the log, shouldn't we have the same behavior accross all the logging driver that supports `tag`s ?

Comment (thaJeztah) [inline daemon/logger/journald/journald.go]: Agreed, it should be consistent

Comment (bobrik) [inline daemon/logger/journald/journald.go]: Looks like it was [inconsistent already](https://github.com/docker/docker/blob/a31435c8a2c4560b6e1a0e84450c45490fa587be/daemon/logger/gelf/gelf.go#L58-L62)

Comment (vdemeester) [inline daemon/logger/journald/journald.go]: @bobrik ah, the one I didn't check :sweat_smile:

File: docs/reference/logging/journald.md (hunk 3 above)
Comment (bobrik) [inline docs/reference/logging/journald.md]: Take a look at other splunk option descriptions, they are not always sentences. "Some markup" sounds ambiguous too.

I can change it to: "Specify template to set `CONTAINER_TAG` value in journald logs. Refer to
[log tag option documentation](log_tags.md) for customizing the log tag format."

File: daemon/logger/journald/journald.go (hunk 2 above)
Comment (bobrik) [inline daemon/logger/journald/journald.go]: Let me know when you make one. I wonder if we even need second argument if it's always the same for all loggers.

```
daemon/logger/fluentd/fluentd.go:	tag, err := loggerutils.ParseLogTag(ctx, "docker.{{.ID}}")
daemon/logger/gelf/gelf.go:	tag, err := loggerutils.ParseLogTag(ctx, "")
daemon/logger/journald/journald.go:	tag, err := loggerutils.ParseLogTag(ctx, "")
daemon/logger/splunk/splunk.go:	tag, err := loggerutils.ParseLogTag(ctx, "{{.ID}}")
daemon/logger/syslog/syslog.go:	tag, err := loggerutils.ParseLogTag(ctx, "{{.ID}}")
```

File: docs/reference/logging/journald.md (hunk 3 above)
Comment (vdemeester) [inline docs/reference/logging/journald.md]: @bobrik sounds good to me :wink:.

Comment (bobrik) [inline docs/reference/logging/journald.md]: Updated

**Comment by vdemeester** [comment]
ah @bobrik needs a rebase :sweat:

**Comment by bobrik** [comment]
Did a rebase, but didn't see any conflicts.

**Comment by vdemeester** [comment]
LGTM 🐰

**Pull request merged**